src/Hadolint/Rules.hs

```diff
@@ -737,3 +737,15 @@ usePipefail = instructionRuleState code severity message check False
             , arg <- Bash.getAllArgs cmd
             , arg == "pipefail"
             ]
+
+allowedRegistry :: Set.Set Registry -> Rule
+allowedRegistry allowed = instructionRule code severity message check
+  where
+    code = "DL3026"
+    severity = ErrorC
+    message = "Only use one of the allowed registries in the FROM image"
```

Comment:
`Use only one ...` sounds better to me, if you do not mind.

Anyway, does it really allow only one registry from the list of registries? Why?

Reply:
Well, you can only use one registry per `FROM` instruction. Maybe I should rename `one` with `an`:

`Only use an allowed registries in the FROM image`

Reply:
You think is somebody will use `evil.io/docker.io/image`  with `--trusted-registry docker.io` and we will say it is ok because it is docker.io?

Reply:
@zemanlx It's just bad wording, I pushed a change with something that I think reads better